lib/providers/share/share.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+import 'package:guam_community_client/commons/guam_progress_indicator.dart';
+import 'package:guam_community_client/models/boards/post.dart';
+import 'package:guam_community_client/providers/messages/messages.dart';
+import 'package:guam_community_client/providers/posts/posts.dart';
+import 'package:guam_community_client/providers/user_auth/authenticate.dart';
+import 'package:guam_community_client/screens/boards/posts/detail/post_detail.dart';
+import 'package:guam_community_client/mixins/toast.dart';
+import 'package:provider/provider.dart';
+import 'package:share_plus/share_plus.dart' as share_plus;
+
+class Share with Toast {
+  BuildContext context;
+  bool isLoading = false;
+  Share(this.context);
+
+  Future<bool> initialize() async {
+    bool dynamicLinkExists = await _getInitialLink();
+    _addListener();
+
+    return dynamicLinkExists;
+  }
+
+  Future<bool> _getInitialLink() async {
+    final PendingDynamicLinkData? initialLink = await FirebaseDynamicLinks.instance.getInitialLink();
+    if (initialLink != null) {
+      _navigateScreen(initialLink.link.path);
+      return true;
+    }
+
+    return false;
+  }
+
+  void _addListener() {
+    FirebaseDynamicLinks.instance.onLink.listen((dynamicLinkData) {
+      _navigateScreen(dynamicLinkData.link.path);
+    }).onError((e) {
+      print(e);
+    });
+  }
+
+  void _navigateScreen(String dynamicLink) {
+    Authenticate authProvider = context.read<Authenticate>();
+    Posts postProvider = Posts(authProvider);
+    String postId = dynamicLink.split('/').last;
+    // 페이지 이동
+    Navigator.of(context).push(
```

Comment:
iOS에서는 DynamicLink를 통해 app으로 redirect되면, 하단 네비게이션 바가 사라지고 **CommonTextField** 까지만 렌더링됩니다. Android에서도 그런가요? 그런 경우, rootNavigator를 true로 설정해주면 해결될 듯합니다.

```suggestion
    Navigator.of(context, rootNavigator: true).push(
```